src/components/Form.js

```diff
@@ -82,11 +82,18 @@ export default class Form extends Component {
       const {errors, errorSchema} = this.validate(formData);
       state = {...state, errors, errorSchema};
     }
-    setState(this, state, () => {
+
+    if ("formData" in this.props) {
       if (this.props.onChange) {
-        this.props.onChange(this.state);
+        this.props.onChange(state);
       }
-    });
+    } else {
+      setState(this, state, () => {
+        if (this.props.onChange) {
+          this.props.onChange(this.state);
+        }
+      });
```

Comment:
We'll have to document this behavior very explicitly, and early in the docs, as it may be very surprising to users. Otherwise it makes sense, we can now distinguish between controlled and uncontrolled forms.